### PR TITLE
feat: add timed_out status to managed proxy workflow

### PR DESCRIPTION
**Pull request by frankh**
## Problem

currently if a user creates a proxy but never sets the DNS records it eventually transitions to "erroring", which is not helpful

instead set it to its own state, with a help message to explain what to do

TODO: add a "retry" button instead of telling user to delete and recreate

image

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## How did you test this code?

ran workflows locally

### CODE REVIEW

**Review by benjackwhite — APPROVED** [review]
geenrally looks good to me

File: frontend/src/scenes/settings/project/ManagedReverseProxy.tsx [hunk 1]
```diff
@@ -25,6 +25,16 @@ import { proxyLogic, ProxyRecord } from './proxyLogic'
...
 
+function StatusText(status: string): string {
+    if (status === 'valid') {
+        return 'live'
+    }
+    if (status === 'timed_out') {
+        return 'timed out'
+    }
+    return status
+}
```
Comment (daibhin) [inline frontend/src/scenes/settings/project/ManagedReverseProxy.tsx:36]:
```suggestion
const statusText = {
    'valid': 'live',
    'timed_out': 'timed out'
}
```

File: frontend/src/scenes/settings/project/ManagedReverseProxy.tsx [hunk 2]
```diff
@@ -58,12 +68,17 @@ export function ManagedReverseProxy(): JSX.Element {
                         )}
                     >
                         {status === 'issuing' && <Spinner />}
-                        <span className="capitalize">{status === 'valid' ? 'live' : status}</span>
+                        <span className="capitalize">{StatusText(status)}</span>
```
Comment (daibhin) [inline frontend/src/scenes/settings/project/ManagedReverseProxy.tsx:71]:
```suggestion
                        <span className="capitalize">{statusText[status] || status}</span>
```

**Review by daibhin — APPROVED** [review]

File: posthog/models/proxy_record.py [hunk 3]
```diff
@@ -17,6 +17,7 @@ class Status(models.TextChoices):
         VALID = "valid"
         ERRORING = "erroring"
         DELETING = "deleting"
+        TIMED_OUT = "timed_out"
```
Comment (daibhin) [inline posthog/models/proxy_record.py:20]: Is there a way to get back out of the timed out state if you add the DNS states at a later date?

**Review by daibhin — APPROVED** [review]

**Review by benjackwhite — APPROVED** [review]
Some ci failures but other than that looks about right 🤷

**Pull request merged**